File: src/main/java/com/chattriggers/ctjs/imports/ModuleMetadata.java
package com.chattriggers.ctjs.imports;

import lombok.Getter;

import java.util.ArrayList;

public class ModuleMetadata {
    @Getter
    private String name;
    @Getter
    private int version;
    @Getter
    private ArrayList<String> tags;
    @Getter
    private String pictureLink;
    @Getter
    private String creator;
    @Getter
    private String description;

    public ModuleMetadata(String name, String creator, String description, int version, ArrayList<String> tags, String pictureLink) {
        this.name = name;
        this.creator = creator;
        this.description = description;
        this.version = version;
        this.tags = tags;
        this.pictureLink = pictureLink;
    }
}

File: src/main/kotlin/com/chattriggers/ctjs/engine/module/ModuleMetadata.kt
package com.chattriggers.ctjs.engine.module

import java.util.*

data class ModuleMetadata(
    val name: String? = null,
    val version: String? = null,
    var entry: String? = null,
    val asmEntry: String? = null,
    val asmExposedFunctions: Map<String, String>? = null,
    val tags: ArrayList<String>? = null,
    val pictureLink: String? = null,
    val creator: String? = null,
    val description: String? = null,
    val requires: ArrayList<String>? = null,
    val helpMessage: String? = null,
    val changelog: String? = null,
    val ignored: ArrayList<String>? = null,
    var isRequired: Boolean = false
) {
    val isDefault: Boolean
        get() = name == null
}
